### PR TITLE
:bug: Source Square migrate auth spec to advanced auth

### PR DESCRIPTION
**Pull request by davydov-d**
## What
Remove deprecated authSpecification from connector spec in favour of advancedAuth
https://github.com/airbytehq/airbyte/issues/26247
+fix build

## How
Replace `authSpecification` with `advancedAuth`

Pay attention: this PR could not be manually tested on localhost because the Square OAuth API does not accept a redirect URL, it is hardcoded in our Square account as `cloud.airbyte.io`

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]
LGTM, once the CAT passes.

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-square/acceptance-test-config.yml [hunk 1]
```diff
@@ -42,6 +42,11 @@ acceptance_tests:
             bypass_reason: "Floating data"
           - name: updated_at
             bypass_reason: "Floating data"
+        categories:
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-square/acceptance-test-config.yml:45]: How is this related to the oauth change?

File: airbyte-integrations/connectors/source-square/source_square/spec.yaml [hunk 2]
```diff
@@ -75,17 +75,6 @@ connectionSpecification:
       title: Include Deleted Objects
       default: false
       order: 3
-authSpecification:
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-square/source_square/spec.yaml:78]: I have the same question as [this other PR](https://github.com/airbytehq/airbyte/pull/26358/files#r1204215234)

**Review by maxi297 — APPROVED** [review]
Conditional approval on the confirmation that this isn't breaking change

File: airbyte-integrations/connectors/source-square/acceptance-test-config.yml (hunk 1 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-square/acceptance-test-config.yml:45]: This was needed to fix buid, not actually related to oauth changes
